### PR TITLE
feat: add remote toggle for livechat and whatsapp

### PR DESCRIPTION
**Pull request by henry-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/javascript/app/base/header.js [hunk 1]
```diff
@@ -60,8 +61,38 @@ const Header = (() => {
...
+    const applyFeatureFlags = () => {
+        getRemoteConfig(true)
+            .then(data => {
+                const { cs_chat_livechat, cs_chat_whatsapp } = data.data;
+                const mobile_menu_livechat                   = getElementById('mobile__menu-livechat');
+                const livechat                               = getElementById('livechat');
+                const topbar_whatsapp                        = getElementById('topbar-whatsapp');
+                const whatsapp_mobile_drawer                 = getElementById('whatsapp-mobile-drawer');
+
+                if (cs_chat_livechat) {
```
Comment (kate-deriv) [inline src/javascript/app/base/header.js]: I think this part can be improved, as the logic with applying inline-flex and none is similar.
 Something like this:

```
 const updateVisibility = ({condition, element_1, element_2}) => {
    const display_type= condition ? 'inline-flex' : 'none';
    element_1.style.display = display_type;
    element_2.style.display = display_type;
}

updateVisibility({condition: cs_chat_livechat, element_1: mobile_menu_livechat, element_2: livechat});
updateVisibility({condition: cs_chat_whatsapp, element_1: topbar_whatsapp, element_2: whatsapp_mobile_drawer});
```

File: src/javascript/app/base/header.js [hunk 2]
```diff
@@ -40,7 +41,7 @@ const Header = (() => {
         fnc_enter: ['requestFullscreen', 'webkitRequestFullscreen', 'mozRequestFullScreen', 'msRequestFullscreen'],
         fnc_exit : ['exitFullscreen', 'webkitExitFullscreen', 'mozCancelFullScreen', 'msExitFullscreen'],
     };
-
+
```
Comment (kate-deriv) [inline src/javascript/app/base/header.js]: Nit: empty space

File: src/javascript/app/hooks/useRemoteConfig.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        } catch (error) {
+            // eslint-disable-next-line no-console
+            console.log('Remote Config error: ', error);
+        }
+        return data;
+    };
+
+    const getRemoteConfig = async (enabled = false) => {
+        isEnabled = enabled;
+        if (isEnabled) {
```
Comment (kate-deriv) [inline src/javascript/app/hooks/useRemoteConfig.js:40]: Not sure, but will it work?  
```
const getRemoteConfig = async (enabled = false) => {
        if (enabled) await fetchAndUpdateData();
        return { data };
}
```

Comment (henry-deriv) [reply src/javascript/app/hooks/useRemoteConfig.js:40]: can but i still have to update isEnabled. hence was using isEnabled.

Comment (kate-deriv) [reply src/javascript/app/hooks/useRemoteConfig.js:40]: aaa, okay, got it, then ignore me comment please

**Review by kate-deriv — APPROVED** [review]

**Review by akmal-deriv — APPROVED** [review]

**Review by nijil-deriv — APPROVED** [review]

**Review by ahmadtaimoor-deriv — APPROVED** [review]

**Pull request merged**